src/test/java/StringManipulationTest.java

```diff
@@ -55,27 +69,46 @@ public void testRemoveNthCharacter2() {
 
     @Test
     public void testRemoveNthCharacter3() {
-        fail("Not yet implemented");
+    	//checks if IllegalArgumentException is thrown when n is less than zero
+    	manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+    	IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class, () -> {
+    		assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(-1, true));
+    	  }, "n must be at least 1.");
+        assertEquals("n must be at least 1.", thrown.getMessage());
     }
 
     @Test
     public void testRemoveNthCharacter4() {
-        fail("Not yet implemented");
+    	//checks if indexoutofboundsexception is thrown when n is greater than string length
+    	manipulatedstring.setString("I'd b3tt3r put s0me d161ts in this 5tr1n6, right?");
+    	IndexOutOfBoundsException thrown = assertThrows(IndexOutOfBoundsException.class, () -> {
+    		assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(50, true));
+    	  }, "index out of bounds");
+        assertEquals("index out of bounds", thrown.getMessage());
     }
 
     @Test
     public void testRemoveNthCharacter5() {
-        fail("Not yet implemented");
-    }
+    	//checks if IllegalArgumentException is thrown when n is zero
+    	manipulatedstring.setString("the quick brown fox jumps over the lazy dog");
+    	IllegalArgumentException thrown = assertThrows(IllegalArgumentException.class, () -> {
+    		assertEquals("I'  b tt r  ut s0 e  16 ts in th s  tr n6  r gh ?", manipulatedstring.removeNthCharacter(0, true));
+    	  }, "n must be at least 1.");
+    	assertEquals("n must be at least 1.", thrown.getMessage());
+    }    
 
     @Test
     public void testRemoveNthCharacter6() {
-        fail("Not yet implemented");
+    	//blanks out string, maintainSpacing=true
+    	manipulatedstring.setString("the quick brown fox jumps over the lazy dog");
+        assertEquals("                                           ", manipulatedstring.removeNthCharacter(1, true));
     }
 
     @Test
     public void testRemoveNthCharacter7() {
-        fail("Not yet implemented");
+    	//blanks out string, maintainSpacing=false
+    	manipulatedstring.setString("the quick brown fox jumps over the lazy dog");
+        assertEquals("", manipulatedstring.removeNthCharacter(1, false));
     }
```

Comment:
6 and 7 are good tests, it is worth testing to verify a behavior that you'd expect like this when n = 1.